packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit.tsx

```diff
@@ -86,72 +102,270 @@ const BatchRow: FC<{ batch: InboundShipmentRow }> = ({ batch }) => {
+    <Table>
+      <TableHead>
+        <TableRow>
+          <HeaderCell />
+          <HeaderCell>{t('label.batch')}</HeaderCell>
+          <HeaderCell align="right">{t('label.num-packs')}</HeaderCell>
+          <HeaderCell>Volume per Pack</HeaderCell>
+          <HeaderCell>Weight per Pack</HeaderCell>
+          <HeaderCell align="right">Line total weight</HeaderCell>
+          <HeaderCell align="right">Line total volume</HeaderCell>
+        </TableRow>
+      </TableHead>
+
+      {batches.map((batch, index) => (
+        <WeightsBatchRow
+          label={t('label.line', {
+            line: index + 1,
+          })}
+          key={batch.id}
+          batch={batch}
+        />
+      ))}
+    </Table>
+  );
+};
+
+const PricingBatchRow: FC<{ batch: InboundShipmentRow; label: string }> = ({
+  batch,
+  label,
+}) => {
+  return (
+    <TableRow sx={{ height: 40 }}>
+      <BasicCell>{label}</BasicCell>
+      <EditableCell
+        onChange={newValue => batch.update?.('batch', newValue)}
+        value={batch.batch}
+      />
+
       <BasicCell>
-        <NumericTextInput width={30} value={null} onChange={() => {}} />
+        <NumericTextInput
+          width={125}
+          value={batch.sellPricePerPack}
+          onChange={e =>
+            batch.update?.('sellPricePerPack', Number(e.target.value))
+          }
+        />
       </BasicCell>
+
       <BasicCell>
-        <NumericTextInput width={30} value={null} onChange={() => {}} />
+        <NumericTextInput
+          width={125}
+          value={batch.costPricePerPack}
+          onChange={e =>
+            batch.update?.('costPricePerPack', Number(e.target.value))
+          }
+        />
       </BasicCell>
       <BasicCell>
-        <NumericTextInput width={30} value={null} onChange={() => {}} />
+        <NumericTextInput width={125} value={null} onChange={() => {}} />
+      </BasicCell>
+      <BasicCell align="right" sx={{ width: 125 }}>
+        {batch.numberOfPacks * batch.packSize * batch.costPricePerPack}
       </BasicCell>
 
       <td style={{ marginBottom: 10 }} />
     </TableRow>
   );
 };
 
+const PricingTable: FC<{ batches: InboundShipmentRow[] }> = ({ batches }) => {
+  const t = useTranslation(['outbound-shipment', 'common']);
+  return (
+    <Table>
+      <TableHead>
+        <TableRow>
+          <HeaderCell />
+          <HeaderCell>{t('label.batch')}</HeaderCell>
+          <HeaderCell align="right">{t('label.sell')}</HeaderCell>
+          <HeaderCell align="right">{t('label.cost')}</HeaderCell>
+          <HeaderCell align="right">% Margin</HeaderCell>
+          <HeaderCell align="right">Line Total</HeaderCell>
+        </TableRow>
+      </TableHead>
+
+      {batches.map((batch, index) => (
+        <PricingBatchRow
+          label={t('label.line', {
+            line: index + 1,
+          })}
+          key={batch.id}
+          batch={batch}
+        />
+      ))}
+    </Table>
+  );
+};
+
+enum Tabs {
+  General = 'General',
```

Comment:
Batch? Quantity? Just wondering if we can get away from general :)

Guessing that there will be some opinions about names 😉  What you've done is sensible and clear to me.
Translations, when we go ahead with it.